jetbrains-core/src/software/aws/toolkits/jetbrains/core/tools/Versions.kt

```diff
@@ -0,0 +1,44 @@
+// Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.core.tools
+
+/**
+ * Top level interface for different versioning schemes such as semantic version
+ */
+interface Version : Comparable<Version> {
+    /**
+     * @return Human-readable representation of the version
+     */
+    fun displayValue(): String
+}
+
+/**
+ * @return true if the specified version is compatible with any of the specified version ranges. Always returns true if no ranges are specified.
+ */
+fun <T : Version> isVersionValid(version: T, ranges: List<VersionRange<T>>): Validity {
```

Comment:
nit: could prly be an extension function, since `version: T` is the "subject"

Reply:
changed and renamed to `isValid`